src/archivist/index.js

```diff
@@ -182,12 +186,32 @@ export default class Archivist extends events.EventEmitter {
   }
 
   async extractVersionContent(sourceDocuments) {
-    return (await Promise.all(sourceDocuments.map(async sourceDocument => this.extract(sourceDocument)))).join(Version.SOURCE_DOCUMENTS_SEPARATOR);
+    const extractDocumentErrors = [];
+
+    const result = await Promise.all(sourceDocuments.map(async sourceDocument => {
+      try {
+        return await this.extract(sourceDocument);
+      } catch (error) {
+        if (!(error instanceof ExtractDocumentError)) {
+          throw error;
+        }
+
+        extractDocumentErrors.push(error.message);
+      }
+    }));
+
+    if (extractDocumentErrors.length) {
+      throw new InaccessibleContentError(extractDocumentErrors);
+    }
```

Comment:
This change is part of a commit named “Consider extraction errors as operational errors”, but I fail to understand exactly what we are adding here. As I understand it, we are now making `extract` potentially fail if any single document fails at extraction 😱 
We should probably have a sync call on that one 🙂

Reply:
I had misread the variable names: this will not fail _tracking_ if any _terms_ fail to extract, but fail the _extraction_ if any _document_ fails to extract, which is right 🙂